trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/model/OutputConfigurationQueryParameters.java

```diff
@@ -0,0 +1,29 @@
+/**********************************************************************
+ * Copyright (c) 2023 Ericsson
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License 2.0 which
+ * accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+package org.eclipse.tracecompass.incubator.internal.trace.server.jersey.rest.core.model;
+
+import org.eclipse.jdt.annotation.NonNull;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+
+/**
+ * Contributes to the model used for TSP swagger-core annotations.
+ */
+@Schema(allOf = ConfigurationQueryParameters.class)
+public interface OutputConfigurationQueryParameters extends ConfigurationQueryParameters {
```

Comment:
A suggestion was made to have typeId, name, and description as explicit fields in a parameters map because these are used by the configuration feature itself, so their use can be described in the TSP, and they don't need to be included or described in the JSON Schema.

For the application-specific parameters, it was suggested to use a 'properties' field in the parameters map, where the value is an object that matches and validates against the corresponding 'properties' definition in the JSON Schema of the configuration source type.

Reply:
It's true, that name and description are valid for all configurations, so is the typeId. They could be part of the TSP definition instead of each configuration extension. The typeId would be required, the name and description can be optional. 

Using properties would help us to distinguish the TSP string parameters and the custom parameters. However, if we use parameters, then the whole json would look like a serialized ITmfConfiguration (without the ID which will be assigned). I think both are ok.

So here how it would look like from the TSP point of view (for InAndOut analysis) :

```json
{
    "parameters": {
        "name": "An InAndOut Analysis",
        "description": "My own InAndOut analysis",
        "typeId": "org.eclipse.tracecompass.incubator.internal.inandout.core.config",
        "properties": 
         {
            "specifiers" : [
              {
                "label":"Latency",
                "inRegex":"(\\S*)_entry",
                "outRegex":"(\\S*)_exit",
                "contextInRegex":"(\\S*)_entry",
                "contextOutRegex":"(\\S*)_exit",
                "classifier":"CPU"
              }
          ]
        }
    },
}
```

FYI @abhinava-ericsson

Reply:
As per discussion here https://github.com/eclipse-tracecompass-incubator/org.eclipse.tracecompass.incubator/pull/70#discussion_r1805105697, properties will be replaced by parameters to match the Configuration data structure

Reply:
I came to the realization, that having the name, description and type ID inside the Map<String, Object> and extracting the custom parameters map is not ideal. The items passed are actually the items of an `ITmfConfiguration` object, where the ID is created in the back-end and doesn't need to be passed.